Tools/MGCB/CommandLineParser.cs

```diff
@@ -284,6 +284,8 @@ private IEnumerable<string> Preprocess(IEnumerable<string> args)
                 if (arg.StartsWith("/@:") || arg.StartsWith("-@:"))
                 {
                     var file = arg.Substring(3);
+                    Directory.SetCurrentDirectory(Path.GetDirectoryName(file));
```

Comment:
I think the correct fix is to expand all path's to absolute when the preprocessor expands a response file. This fixes your issue without messing with the working directory for anyone downstream. Also note that restoring the working directory after the response file could work, so long as it is done in a stack so response files with response files don't break.